network-libp2p/tests/network.rs

```diff
@@ -405,15 +405,33 @@ async fn connections_are_properly_closed() {
     log::debug!("closed peer");
 
     let event1 = events1.next().await.unwrap().unwrap();
-    assert_peer_left(&event1, net2.local_peer_id());
+    assert_peer_left(&event1, &net2.local_peer_id());
     log::trace!(event1 = ?event1);
 
     let event2 = events2.next().await.unwrap().unwrap();
     assert_peer_left(&event2, net1.local_peer_id());
     log::trace!(event2 = ?event2);
+}
+
+#[test(tokio::test)]
+async fn connections_are_properly_closed_peers() {
+    let (net1, net2) = create_connected_networks().await;
+
+    let peer1 = net2.get_peer(*net1.local_peer_id()).unwrap();
+
+    let mut events1 = net1.subscribe_events();
+
+    let net2_peer_id = *net2.local_peer_id();
+    drop(net2);
+
+    peer1.close(CloseReason::Other);
```

Comment:
I'd first close `peer1` connection and then drop `net2` since `peer1` comes from `net2`.

Reply:
I misidentified whose network `peer1` is, I thought it was a peer of `net1`. I now swapped the roles of `net1` and `net2` and it should be good now?